x/hub-genesis/keeper/msg_server.go

```diff
@@ -0,0 +1,82 @@
+package keeper
+
+import (
+	"context"
+	"encoding/json"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	transfertypes "github.com/cosmos/ibc-go/v6/modules/apps/transfer/types"
+	clienttypes "github.com/cosmos/ibc-go/v6/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v6/modules/core/04-channel/types"
+	"github.com/dymensionxyz/dymension-rdk/x/hub-genesis/types"
+	"github.com/dymensionxyz/gerr-cosmos/gerrc"
+)
+
+const port = transfertypes.PortID
+
+var _ types.MsgServer = msgServer{}
+
+type msgServer struct{ Keeper }
+
+func NewMsgServerImpl(keeper Keeper) types.MsgServer {
+	return &msgServer{Keeper: keeper}
+}
+
+// note that this is only allowed by the whitelisted relayer (enforced in ante)
+func (m msgServer) SendTransfer(goCtx context.Context, msg *types.MsgSendTransfer) (*types.MsgSendTransferResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	err := m.SendGenesisTransfer(ctx, msg.ChannelId)
+	if err != nil {
+		return nil, err
+	}
+	return &types.MsgSendTransferResponse{}, nil
+}
+
+func (k Keeper) SendGenesisTransfer(ctx sdk.Context, channelID string) error {
+	state := k.GetState(ctx)
+	if state.InFlight {
+		return gerrc.ErrFailedPrecondition.Wrap("sent transfer is already in flight")
+	}
+	c, ok := k.channelKeeper.GetChannel(ctx, port, channelID)
+	if !ok {
+		return gerrc.ErrNotFound.Wrap("channel")
+	}
+	if c.State != channeltypes.OPEN {
+		return gerrc.ErrFailedPrecondition.Wrap("channel not open")
+	}
+	if !state.CanonicalHubTransferChannelHasBeenSet() {
+		state.SetCanonicalTransferChannel(port, channelID)
```

Comment:
`state.SetCanonicalTransferChannel` should be set on success, not on submission trigger

Reply:
why?

Reply:
There is no need to set another channel
The relayer tells you the right channel first time and that is that

Reply:
ok np
just means that if wrong channel will be set, it's irreversible